deployment/data-streams/jobs/generator.go

```diff
@@ -0,0 +1,65 @@
+package jobs
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// JobSpecGenerator knows how to generate job specs for each stream type.
+// It doesn't cover bootstrap and LLO job specs.
+type JobSpecGenerator interface {
+	GenerateJobSpec(ssc StreamSpecConfig, externalJobID uuid.UUID) (*StreamJobSpec, error)
+}
+
+func GeneratorForStreamType(st StreamType) (JobSpecGenerator, error) {
+	switch st {
+	case StreamTypeQuote:
+		return &QuoteStreamJobSpecGenerator{}, nil
+	// case StreamTypeMedian:
```

Comment:
Basically this is where we add new stream types?

Reply:
Yes, the idea is that we have default generators for each stream type and they will be defined here. We can still pass a custom one via the changeset parameters but normally we'll use the ones defined here.